mm2src/coins/lightning.rs

```diff
@@ -300,20 +510,47 @@ impl SwapOps for LightningCoin {
         _amount: BigDecimal,
         _swap_contract_address: &Option<BytesJson>,
         _swap_unique_data: &[u8],
+        payment_instructions: &Option<PaymentInstructions>,
     ) -> TransactionFut {
-        unimplemented!()
+        let PaymentInstructions::Lightning(invoice) = try_tx_fus!(payment_instructions
+            .clone()
+            .ok_or_else(|| TransactionErr::Plain("payment_instructions can't be None".into())));
+        let coin = self.clone();
+        let fut = async move {
+            let payment = try_tx_s!(coin.pay_invoice(invoice).await);
+            Ok(payment.payment_hash.into())
+        };
+        Box::new(fut.boxed().compat())
     }
 
     fn send_maker_spends_taker_payment(
         &self,
-        _taker_payment_tx: &[u8],
+        taker_payment_tx: &[u8],
         _time_lock: u32,
         _taker_pub: &[u8],
-        _secret: &[u8],
+        secret: &[u8],
         _swap_contract_address: &Option<BytesJson>,
         _swap_unique_data: &[u8],
     ) -> TransactionFut {
-        unimplemented!()
+        let payment_hash = try_tx_fus!(payment_hash_from_slice(taker_payment_tx));
```

Comment:
At first sight this looks like you're taking the hash of the lightning payment bytes `taker_payment_tx`, and that this is something different than the regular `payment_hash` i.e. hash of the payment_preimage. This confused me for a while. Can you change the parameter name `taker_payment_tx` to `payment_hash`? Also in other `SwapOps` methods.

Reply:
>  Can you change the parameter name `taker_payment_tx` to payment_hash?

This can't be changed since in other `SwapOps` implementations the `taker_payment_tx` bytes are what's passed. There is no concept of payment bytes in lightning, this comment can be part of https://github.com/KomodoPlatform/atomicDEX-API/issues/1501